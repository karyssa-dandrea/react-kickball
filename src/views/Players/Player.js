import React from 'react';
import { useState, useEffect } from 'react';
import { fetchPlayerById } from '../../services/players';
import PlayerDetail from './Players';
export default function Player(props) {
  const [player, setPlayer] = useState([]);
  const [loading, setLoading] = useState(true);
  const id = props.match.params.id;

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchPlayerById(id);
      setPlayer(data);
      setLoading(false);
    };
    fetchData();
  }, [id]);

  if (loading) {
    return <h1>Loading</h1>;
  }

  return (
    <div className="player-deets">
      <h1>Players</h1>
      <p>{player.name}</p>
      <PlayerDetail player={player} />
    </div>
  );
}
