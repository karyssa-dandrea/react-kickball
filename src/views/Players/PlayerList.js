import React from 'react';

export default function PlayerList() {
  return <div></div>;
}
