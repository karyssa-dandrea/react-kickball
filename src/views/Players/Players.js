import React from 'react';
import { useState, useEffect } from 'react';
import { fetchPlayers } from '../../services/players';
import PlayerList from '../../components/PlayerFunction/PlayerList';
import './Players.css';

export default function Players() {
  const [players, setPlayers] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchPlayers();
      setPlayers(data);
    };
    fetchData();
  }, []);
  return (
    <div>
      <h1>List of Players</h1>
      <PlayerList players={players} />
    </div>
  );
}
