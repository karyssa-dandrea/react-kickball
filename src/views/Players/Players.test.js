import Players from '../../views/Players/Players';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';

test('should render list of players', async () => {
  const { container } = render(
    <MemoryRouter>
      <Players />
    </MemoryRouter>
  );
  await screen.findByText('Betty Grey');
  expect(container).toMatchSnapshot();
});
