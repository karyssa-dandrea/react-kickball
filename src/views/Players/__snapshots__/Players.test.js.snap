// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render list of players 1`] = `
<div>
  <div>
    <h1>
      List of Players
    </h1>
    <div
      class="players-list"
    >
      <a
        href="/players/2"
      >
        Hans Olo
      </a>
      <a
        href="/players/3"
      >
        Ron Long
      </a>
      <a
        href="/players/4"
      >
        Shorty Biggum
      </a>
      <a
        href="/players/5"
      >
        Gary Coolman
      </a>
      <a
        href="/players/6"
      >
        Terry Cruise
      </a>
      <a
        href="/players/7"
      >
        Bud E. Guy
      </a>
      <a
        href="/players/8"
      >
        Buster Rimes
      </a>
      <a
        href="/players/9"
      >
        Deb E. Downer
      </a>
      <a
        href="/players/10"
      >
        Jayne Phonda
      </a>
      <a
        href="/players/11"
      >
        Kara Whollot
      </a>
      <a
        href="/players/12"
      >
        Betty Grey
      </a>
      <a
        href="/players/13"
      >
        Dwayne Elizondo Mountain Dew Herbert Camacho
      </a>
      <a
        href="/players/14"
      >
        Joe Bauers
      </a>
      <a
        href="/players/15"
      >
        Frito P.
      </a>
      <a
        href="/players/1"
      >
        Bennie Jetts
      </a>
    </div>
  </div>
</div>
`;
